src/mme-app/actionHandlers/defaultMmeProcedureActionHandlers.cpp

```diff
@@ -1195,3 +1179,152 @@ ActStatus ActionHandlers::default_path_switch_req_handler(ControlBlock& cb)
     return rc;
 }
 
+/***************************************
+* Action handler : default_delete_subs_req_handler
+***************************************/
+ActStatus ActionHandlers::default_delete_subs_req_handler(ControlBlock& cb)
+{
+    log_msg(LOG_DEBUG, "default_delete_subs_req_handler : Entry\n");
+
+    MsgBuffer* msgBuf = static_cast<MsgBuffer*>(cb.getMsgData());
+    VERIFY(msgBuf, return ActStatus::PROCEED, "Invalid message buffer\n");
+
+    const dsr_Q_msg_t* msgData_p = static_cast<const dsr_Q_msg_t *>(msgBuf->getDataPointer());
+
+    DigitRegister15 IMSI;
+    IMSI.setImsiDigits((unsigned char *)msgData_p->header.IMSI);
+
+    s6a_dsa_Q_msg dsa;
+    dsa.msg_type = delete_subscriber_data_answer;
+    dsa.eteId = msgData_p->eteId;
+    dsa.res_code = DIAMETER_UNABLE_TO_COMPLY;
+
+    UEContext *ueCtxt = static_cast<UEContext*>(cb.getPermDataBlock());
+    if (ueCtxt != NULL)
+    {
+        MmContext* mmCtxt = ueCtxt->getMmContext();
+        if(mmCtxt)
+        {
+            if (msgData_p->dsr_flags & dsrCompApnProfWdwl_c)
+            {
+                log_msg(LOG_INFO, "DSR received with Complete APN Config "
+                    "Profile Withdrawal Flag set to 1");
```

Comment:
what should be the MME action in this complete withdrawal case ?? we should do the idle check for this also right ?

Reply:
In 29.272, it is stated that , 

"The HSS **shall not set** the "**Complete APN Configuration Profile Withdrawal"** bit in the DSR-Flags AVP when sending a Delete Subscriber Data Request to an MME, since the default APN shall always be present in an MME. "

"If an MME receives a Delete Subscriber Data Request with the "Complete APN Configuration Profile Withdrawal" bit set in the DSR-Flags AVP, it shall return an error with a Result-Code set to DIAMETER_UNABLE_TO_COMPLY. "

So, idle check is not needed here. We are responding with **DIAMETER_UNABLE_TO_COMPLY** and we shall change the log level to "LOG_ERROR".